tests/testthat/test-SQLServer.R

```diff
@@ -173,6 +173,26 @@ test_that("SQLServer", {
     expect_equal(as.double(values[[6]]), as.double(received[[6]]))
   })
 
+  local({
+    con <- DBItest:::connect(DBItest:::get_default_context())
+    input <- DBI::SQL(c(
+               "testtable",
+               "[testtable]",
+               "[testschema].[testtable]",
+               "[testschema].testtable",
+               "[testdb].[testschema].[testtable]",
+               "[testdb].[testschema].testtable" ))
+    expected <- c(
+                  DBI::Id( table = "testtable" ),
+                  DBI::Id( table = "testtable" ),
+                  DBI::Id( schema = "testschema", table = "testtable" ),
+                  DBI::Id( schema = "testschema", table = "testtable" ),
+                  DBI::Id( catalog = "testdb", schema = "testschema", table = "testtable" ),
+                  DBI::Id( catalog = "testdb", schema = "testschema", table = "testtable" ) )
+    expect_identical( DBI::dbUnquoteIdentifier( con, input ), expected )
```

Comment:
I'm a bit surprised that `dbUnquoteIdentifier` returns an `Id`, but that's a DBI issue, and too late to change. (Although the docs could be better)